examples/google-analytics/app/utils/gtags.ts

```diff
@@ -0,0 +1,32 @@
+export const GA_TRACKING_ID = "Your google analytics id goes here.";
+
+declare global {
+  interface Window {
+    gtag: (
+      option: string,
+      gaTrackingId: string,
+      options: Record<string, unknown>
+    ) => void;
+  }
+}
+
+// https://developers.google.com/analytics/devguides/collection/gtagjs/pages
+export const pageview = (url: string) => {
+  window.gtag("config", GA_TRACKING_ID, {
+    page_path: url
+  });
+};
+
+// https://developers.google.com/analytics/devguides/collection/gtagjs/events
+export const event = ({
+  action,
+  category,
+  label,
+  value
+}: Record<string, string>) => {
+  window.gtag("event", action, {
+    event_category: category,
+    event_label: label,
+    value: value
+  });
+};
```

Comment:
I would consider renaming this file to `gtags.client.ts` since it can only run in the browser because of usage of the `window` object.

I was getting errors on the server when importing this module in `root.tsx`, but adding `client` as part of the name in this file  fixed my problem